data/campaigns/Heir_To_The_Throne/scenarios/07_Crossroads.cfg

```diff
@@ -321,46 +321,55 @@
             # variables, thus always having the same ratio of different
             # types.
 
-            [unit]
-                x,y=$ambush_location.x,$ambush_location.y
-                type=$random_ambusher_type_table[$random_ambusher_type_i].type
-                side={SIDE}
-                facing=sw
-                role=ambusher
-                random_traits=yes
-
-                [modifications]
-                    {TRAIT_LOYAL}
-                [/modifications]
-                {IS_LOYAL}
-            [/unit]
-
-            [object]
-                silent=yes
-
-                [filter]
-                    x,y=$ambush_location.x,$ambush_location.y
-                [/filter]
-
-                [effect]
-                    apply_to=new_ability
-
-                    [abilities]
-                        [hides]
-                            id=orc_ambush
-                            affect_self=yes
-
-                            [filter_self]
-                                role=ambusher
-
-                                [filter_location]
-                                    terrain=Hh,Mm
-                                [/filter_location]
-                            [/filter_self]
-                        [/hides]
-                    [/abilities]
-                [/effect]
-            [/object]
+            [if]
+                [not]
+                    [have_unit]
+                        x,y=$ambush_location.x,$ambush_location.y
+                    [/have_unit]
+                [/not]
```

Comment:
There is a lot of legacy cruft. I didn't get into what could be improved, only what needed fixin'

Otherwise, I'd probably have re-written most of the campaign. In fact, I did re-write some but decided, when rolling this series of PRs to eliminate it because it was mainly gratuitous and didn't solve any real problems other than utter un-readability.

That said, if there are areas which you want re-written, sure, I'll take 'em on.